source/deployment/cluster.rst

```diff
@@ -0,0 +1,207 @@
+                proxy_set_header Upgrade $http_upgrade;
+              proxy_set_header Connection "upgrade";
+                proxy_set_header Host $http_host;
+                proxy_set_header X-Real-IP $remote_addr;
+                proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
+                proxy_set_header X-Forwarded-Proto $scheme;
+                proxy_set_header X-Frame-Options SAMEORIGIN;
+                proxy_pass http://backend;
+          }
+    }
+
+
+A setup with multiple proxy servers can be utilized to limit a single point of failure, but is beyond the scope of this documentation.
+
+
+File Storage Configuration
+----------------------------------------
+
+.. note:: File storage is assumed to be shared between all the machines utilizing services such as NAS or Amazon S3. If ``"DriverName": "local"`` is used then the directory at ``"FileSettings":`` ``"Directory": "./data/"`` is expected to be a NAS location mapped as a local directory, otherwise high availability will not function correctly and may corrupt your file storage. If you’re using Amazon S3 for file storage then no other configuration is required.
+
+If you’re using the Compliance Reports feature in Enterprise Edition E20, you will need to configure the  ``"ComplianceSettings":`` ``"Directory": "./data/",`` to share between all machines or the reports may not be accessible from the System Console and only available on the local Mattermost server.
+
+Migrating to NAS or S3 from local storage is beyond the scope of this document.
+
+Database Configuration
+------------------------------------
+Scaling the database can be accomplished by utilizing the read-replica feature. The Mattermost server can be set up to use one "master" database and up to 8 read replica databases. Mattermost distributes read requests across all databases, and sends write requests to the master database, and those changes are then sent to update the read replicas. 
+
+Sizing databases
+```````````````````````````````````````
+Please see `documentation on sizing database servers <http://docs.mattermost.com/install/requirements.html#hardware-requirements>`_ for guidance to determine appropriate hardware. 
+
+It's highly recommended that at least one read replica have a minimum of **three times the disk space** of the master database. This way, if the master database signals a "low disk space" warning the administrator can switch over to the read replica with a larger disk and continue operation, while also having room to locally duplicate and manipulate the production database. 
+
+
+Deploying a multi-database configuration 
+````````````````````````````````````````````````````````````````````````
+To configure a multi-database Mattermost server: 
+
+1. Update the ``DataSource`` setting ``config.json`` with a connection string to your master database server based on the database type set in ``DriverName``, which would be either ``postgres`` or ``mysql``. 
```

Comment:
Only 2/5, but easier to read if there was a period in between. Maybe

`Update the ``DataSource`` setting in ``config.json`` with a connection string to your master database server. The connection is based on the database type set in DriverName, either ``postgres`` or ``mysql.`